show/bgp_quagga_v4.py

```diff
@@ -14,13 +79,72 @@ def bgp():
     """Show IPv4 BGP (Border Gateway Protocol) information"""
     pass
 
-
+#
+# 'bgp' group ("show ip bgp ...")
+#
+@bgp.group(cls=AliasedGroup, default_if_no_args=False, invoke_without_command=True)
+@click.pass_context
 # 'summary' subcommand ("show ip bgp summary")
-@bgp.command()
-def summary():
-    """Show summarized information of IPv4 BGP state"""
-    run_command('sudo vtysh -c "show ip bgp summary"')
+def summary(ctx):
+    if ctx.invoked_subcommand is None:
+        """Show summarized information of IPv4 BGP state"""
+        run_command('sudo vtysh -c "show ip bgp summary"')
+
+# 'enhanced' subcommand ("show ip bgp summary enhanced")
+@summary.command()
+def enhanced():
+    """Show BGP summary with Devicename and advertised-routes"""
+    # get bgp summary table from cli and save as dictionary
+    bgpsum_cmd = 'sudo vtysh -c "show ip bgp summary"'
+    bgpsum_output = run_command_save(bgpsum_cmd)
+    parse_bgp_sum_output = parse_bgp_summary(bgpsum_output)
+    # only preform the following if there is data from BGP summary
+    if len(parse_bgp_sum_output) > 0:
+        bgpsum_dict = {}
+        for item in parse_bgp_sum_output:
+            bgpsum_dict[item['Neighbor']] = {'V': item['V'],
+                                             'AS': item['AS'],
+                                             'MsgRcvd': item['MsgRcvd'],
+                                             'MsgSent': item['MsgSent'],
+                                             'TblVer': item['TblVer'],
+                                             'InQ': item['InQ'],
+                                             'OutQ': item['OutQ'],
+                                             'Up/Down': item['Up/Down'],
+                                             'State/PfxRcd': item['State/PfxRcd']}
+        # create a list of BGP neighbor
+        neighborIP = bgpsum_dict.keys()
+        neighbor_adv_dict = {}
+        # get advestise prefix per neighbor
+        for item in neighborIP:
+            pfxadv = run_command_save('sudo vtysh -c "show ip bgp nei {0} advertised-routes" | grep Total'.format(item))
```

Comment:
>pfxadv [](start = 12, length = 6)

Since you invoke the command in sequence, it may have time consistence issue. Does it make sense to collect them and finally show in a table?

Reply:
it is indeed the case, the per peer output is stored on neighbor_adv_dict. Then output all togther